lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeNames.cpp

```diff
@@ -873,7 +877,29 @@ std::string SwiftLanguageRuntime::DemangleSymbolAsString(
       return name;
     };
   }
-  return swift::Demangle::demangleSymbolAsString(symbol, options);
+  if (tracking) {
+    TrackingNodePrinter printer = TrackingNodePrinter(options);
+    swift::Demangle::demangleSymbolAsString(symbol, printer);
+    return std::pair<std::string, std::optional<DemangledNameInfo>>(
+        printer.takeString(), printer.takeInfo());
+  }
+  return std::pair<std::string, std::optional<DemangledNameInfo>>(
+      swift::Demangle::demangleSymbolAsString(symbol, options), std::nullopt);
+}
+
+std::string SwiftLanguageRuntime::DemangleSymbolAsString(
+    llvm::StringRef symbol, DemangleMode mode, const SymbolContext *sc,
+    const ExecutionContext *exe_ctx) {
+  return DemangleSymbolAsString(symbol, mode, false, sc, exe_ctx).first;
+}
+
+std::pair<std::string, DemangledNameInfo>
+SwiftLanguageRuntime::TrackedDemangleSymbolAsString(
+    llvm::StringRef symbol, DemangleMode mode, const SymbolContext *sc,
+    const ExecutionContext *exe_ctx) {
+  auto demangledData = DemangleSymbolAsString(symbol, mode, true, sc, exe_ctx);
+  return std::pair<std::string, DemangledNameInfo>(demangledData.first,
+                                                   *demangledData.second);
```

Comment:
Do we need this API? All it does is unwrap the `optional<DemangledNameInfo>`. Could we just let the callers unwrap it?

If we do need this, lets change it to:

```suggestion
  auto [name, info] = DemangleSymbolAsString(symbol, mode, true, sc, exe_ctx);
  return std::pair<std::string, DemangledNameInfo>(std::move(name), std::move(*info));
```

Reply:
Fixed, thanks 👍